From cc455202e8154fd9fe141ede8318a887b58fe8c7 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:49 -0400
Subject: [fs] nfsd: leave reply buffer space for failed setattr

Message-id: <1403560314-32525-48-git-send-email-bfields@redhat.com>
Patchwork-id: 84596
O-Subject: [Patch RHEL7 047/172] nfsd4: leave reply buffer space for failed setattr
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

This fixes an ommission from 18032ca062e621e15683cb61c066ef3dc5414a7b
"NFSD: Server implementation of MAC Labeling", which increased the size
of the setattr error reply without increasing COMPOUND_ERR_SLACK_SPACE.

Cc: stable@vger.kernel.org
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 04819bf6449094e62cebaf5199d85d68d711e667
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfsd.h b/fs/nfsd/nfsd.h
index 30f34ab..479eb68 100644
--- a/fs/nfsd/nfsd.h
+++ b/fs/nfsd/nfsd.h
@@ -282,7 +282,7 @@ void  nfsd_lockd_shutdown(void);
  * reason.
  */
 #define COMPOUND_SLACK_SPACE  140    /* OP_GETFH */
-#define COMPOUND_ERR_SLACK_SPACE 12     /* OP_SETATTR */
+#define COMPOUND_ERR_SLACK_SPACE 16     /* OP_SETATTR */
 
 #define NFSD_LAUNDROMAT_MINTIMEOUT      1   /* seconds */
 
-- 
1.7.1